cmd/airflow.go

```diff
@@ -554,14 +559,80 @@ func airflowInit(cmd *cobra.Command, args []string) error {
 	}
 
 	if exists {
-		fmt.Printf(configReinitProjectConfigMsg+"\n", config.WorkingPath)
+		fmt.Printf(configReinitProjectConfigMsg, config.WorkingPath)
 	} else {
-		fmt.Printf(configInitProjectConfigMsg+"\n", config.WorkingPath)
+		fmt.Printf(configInitProjectConfigMsg, config.WorkingPath)
+	}
+
+	// If we started in a different directory, that means the positional argument for projectName was used.
+	// This means the users shell pwd is not the project directory, so we print a message
+	// to cd into the project directory.
+	if initialDir != config.WorkingPath {
+		fmt.Printf(changeDirectoryMsg, projectName)
 	}
 
 	return nil
 }
 
+// ensureProjectDirectory creates a new project directory if a positional argument is provided.
+func ensureProjectDirectory(args []string, workingPath, projectName string) (string, error) {
+	// Return early if no positional argument was provided.
+	if len(args) == 0 {
+		return workingPath, nil
+	}
+
+	// Construct the path to our desired project directory.
+	newProjectPath := filepath.Join(workingPath, projectName)
+
+	// Determine if the project directory already exists.
+	projectDirExists, err := fileutil.Exists(newProjectPath, nil)
+	if err != nil {
+		return "", err
+	}
+
+	// If the project directory does not exist, create it.
+	if !projectDirExists {
+		err := os.Mkdir(newProjectPath, os.FileMode(directoryPermissions))
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// Return the path we just created.
+	return newProjectPath, nil
+}
+
+func ensureProjectName(args []string, projectName string) (string, error) {
+	// If the project name is specified with the --name flag,
+	// it cannot be specified as a positional argument as well, so return an error.
+	if projectName != "" && len(args) > 0 {
+		return "", errConfigProjectNameSpecifiedTwice
+	}
```

Comment:
@neel-astro added this case here to throw an error message if both the flag and positional argument are used.

I also separated all the new and existing "project name" logic into this new function to make the linter happy. :)